plugins/module_utils/foreman_helper.py

```diff
@@ -494,13 +493,14 @@ def connect(self):
         )
 
         self.status()
+        self.apply_apidoc_patches()
+        self.check_required_plugins()
 
+    def apply_apidoc_patches(self):
```

Comment:
So we turn this hack into a facility now.
This would be the perfect time to add a docstring. ;)

How about:
"Calls some workarounds to fix the local representation of the api docs.
When adding another workaround, consider that the endpoint in question may depend on a plugin to be available.
Also a workaround should be specific to a range of server/plugin versions where possible."

Reply:
Yeah, sounds good.

I wanted to add some more metadata to the patches and make this method more robust (in the future)

Reply:
Do we have an example where such a bug was actually fixed upstream.

Reply:
Yeah, for example https://projects.theforeman.org/issues/10359 was fixed in 1.23, so we won't need to call `_patch_location_api()` on modern installs.

Reply:
Looking forward to see how that works. ;)

Reply:
Used a slightly reworded docstring.